activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb

```diff
@@ -97,10 +97,11 @@ def error_number(exception)
       # QUOTING ==================================================
       #++
 
+      # Quotes strings for use in SQL input.
       def quote_string(string)
-        any_raw_connection.escape(string)
-      rescue Mysql2::Error => error
-        raise translate_exception(error, message: error.message, sql: "<escape>", binds: [])
+        with_raw_connection(allow_retry: true, uses_transaction: false) do |connection|
```

Comment:
I'm assuming `uses_transaction: false` here is what we want, since it seems like we shouldn't need transactions to be materialized in order to do string quoting, but please correct me if I'm wrong.

Reply:
That's correct